### PR TITLE
Update Guzzle to 6.3

### PR DESCRIPTION
**Pull request by phil-davis**
For ref: https://github.com/guzzle/guzzle/blob/master/UPGRADING.md

### CODE REVIEW

File: tests/TestHelpers/EmailHelper.php [hunk 1]
```diff
@@ -39,9 +40,10 @@ class EmailHelper {
 	 */
 	public static function getEmails($mailhogUrl) {
 		$client = new Client();
-		$options = ['headers' => ['Content-Type' => 'application/json']];
-		$request = $client->createRequest(
-			'GET', $mailhogUrl . "/api/v2/messages", $options
+		$request = new Request(
```
Comment (phil-davis) [inline tests/TestHelpers/EmailHelper.php]: ToDo: why is this not calling HttpRequestHelper ?

File: tests/TestHelpers/EmailHelper.php [hunk 2]
```diff
@@ -80,8 +82,9 @@ public static function getBodyOfLastEmail($mailhogUrl, $address) {
 	 */
 	public static function deleteAllMessages($mailhogUrl) {
 		$client = new Client();
-		$request = $client->createRequest(
-			'DELETE', $mailhogUrl . "/api/v1/messages"
+		$request = new Request(
```
Comment (phil-davis) [inline tests/TestHelpers/EmailHelper.php]: ToDo: why is this not calling HttpRequestHelper ?

File: tests/TestHelpers/EmailHelper.php (hunk 1 above)
Comment (phil-davis) [reply tests/TestHelpers/EmailHelper.php]: It is in ``master`` like this. PR #32539 refactors it.

File: tests/TestHelpers/EmailHelper.php (hunk 2 above)
Comment (phil-davis) [reply tests/TestHelpers/EmailHelper.php]: It is in ``master`` like this. PR #32539 refactors it.

**Comment by phil-davis** [comment]
The only remaining fail is in a phpunit with sqlite job:
https://drone.owncloud.com/owncloud/core/10003/166

```
Time: 12.59 minutes, Memory: 2552.00MB

There were 7 errors:

1) Test\Http\Client\ClientTest::testGet
Error: Call to undefined method Mock_Client_56d2bb18::getConfig()

/drone/src/lib/private/Http/Client/Client.php:65
/drone/src/lib/private/Http/Client/Client.php:136
/drone/src/tests/lib/Http/Client/ClientTest.php:93

2) Test\Http\Client\ClientTest::testPost
Error: Call to undefined method Mock_Client_56d2bb18::getConfig()

/drone/src/lib/private/Http/Client/Client.php:65
/drone/src/lib/private/Http/Client/Client.php:202
/drone/src/tests/lib/Http/Client/ClientTest.php:99

3) Test\Http\Client\ClientTest::testPut
Error: Call to undefined method Mock_Client_56d2bb18::getConfig()

/drone/src/lib/private/Http/Client/Client.php:65
/drone/src/lib/private/Http/Client/Client.php:241
/drone/src/tests/lib/Http/Client/ClientTest.php:105

4) Test\Http\Client\ClientTest::testDelete
Error: Call to undefined method Mock_Client_56d2bb18::getConfig()

/drone/src/lib/private/Http/Client/Client.php:65
/drone/src/lib/private/Http/Client/Client.php:276
/drone/src/tests/lib/Http/Client/ClientTest.php:111

5) Test\Http\Client\ClientTest::testOptions
Error: Call to undefined method Mock_Client_56d2bb18::getConfig()

/drone/src/lib/private/Http/Client/Client.php:65
/drone/src/lib/private/Http/Client/Client.php:311
/drone/src/tests/lib/Http/Client/ClientTest.php:117

6) OCA\Files_Sharing\Tests\WatcherTest::testFolderSizePropagationToOwnerStorage
TypeError: Argument 1 passed to Aws\Handler\GuzzleV5\GuzzleHandler::Aws\Handler\GuzzleV5\{closure}() must be an instance of Exception, instance of TypeError given, called in /drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/RejectedPromise.php on line 40

/drone/src/apps/files_primary_s3/vendor/aws/aws-sdk-php/src/Handler/GuzzleV5/GuzzleHandler.php:82
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/RejectedPromise.php:40
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/TaskQueue.php:47
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/Promise.php:246
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/Promise.php:223
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/Promise.php:267
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/Promise.php:225
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/Promise.php:62
/drone/src/apps/files_primary_s3/vendor/aws/aws-sdk-php/src/AwsClientTrait.php:58
/drone/src/apps/files_primary_s3/lib/streamwrapper.php:660
/drone/src/apps/files_primary_s3/lib/streamwrapper.php:143
/drone/src/apps/files_primary_s3/lib/streamwrapper.php:875
/drone/src/apps/files_primary_s3/lib/streamwrapper.php:147
/drone/src/apps/files_primary_s3/lib/s3storage.php:151
/drone/src/lib/private/Files/ObjectStore/ObjectStoreStorage.php:271
/drone/src/lib/private/Files/ObjectStore/ObjectStoreStorage.php:300
/drone/src/lib/private/Files/Storage/Common.php:197
/drone/src/lib/private/Files/Storage/Wrapper/Wrapper.php:252
/drone/src/lib/private/Files/Storage/Wrapper/Availability.php:274
/drone/src/lib/private/Files/Storage/Wrapper/Checksum.php:191
/drone/src/lib/private/Files/Storage/Wrapper/Jail.php:251
/drone/src/apps/files_sharing/lib/SharedStorage.php:452
/drone/src/apps/files_sharing/tests/WatcherTest.php:113

7) OCA\Files_Sharing\Tests\WatcherTest::testSubFolderSizePropagationToOwnerStorage
TypeError: Argument 1 passed to Aws\Handler\GuzzleV5\GuzzleHandler::Aws\Handler\GuzzleV5\{closure}() must be an instance of Exception, instance of TypeError given, called in /drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/RejectedPromise.php on line 40

/drone/src/apps/files_primary_s3/vendor/aws/aws-sdk-php/src/Handler/GuzzleV5/GuzzleHandler.php:82
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/RejectedPromise.php:40
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/TaskQueue.php:47
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/Promise.php:246
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/Promise.php:223
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/Promise.php:267
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/Promise.php:225
/drone/src/apps/files_primary_s3/vendor/guzzlehttp/promises/src/Promise.php:62
/drone/src/apps/files_primary_s3/vendor/aws/aws-sdk-php/src/AwsClientTrait.php:58
/drone/src/apps/files_primary_s3/lib/streamwrapper.php:660
/drone/src/apps/files_primary_s3/lib/streamwrapper.php:143
/drone/src/apps/files_primary_s3/lib/streamwrapper.php:875
/drone/src/apps/files_primary_s3/lib/streamwrapper.php:147
/drone/src/apps/files_primary_s3/lib/s3storage.php:151
/drone/src/lib/private/Files/ObjectStore/ObjectStoreStorage.php:271
/drone/src/lib/private/Files/ObjectStore/ObjectStoreStorage.php:300
/drone/src/lib/private/Files/Storage/Common.php:197
/drone/src/lib/private/Files/Storage/Wrapper/Wrapper.php:252
/drone/src/lib/private/Files/Storage/Wrapper/Availability.php:274
/drone/src/lib/private/Files/Storage/Wrapper/Checksum.php:191
/drone/src/lib/private/Files/Storage/Wrapper/Jail.php:251
/drone/src/apps/files_sharing/lib/SharedStorage.php:452
/drone/src/apps/files_sharing/tests/WatcherTest.php:143

--
```

**Comment by phil-davis** [comment]
ToDo:
Could not share file/folder! message: "cURL error 60: SSL certificate problem: self signed certificate (see http://curl.haxx.se/libcurl/c/libcurl-errors.html)"

Something needs to be looked into related to the new SSL test environment and GuzzleV6.

**Comment by patrickjahns** [comment]
Guzzle is schedule for OC11 - it doesn't make sense to spend much effort here currently.

For Behat/Acceptance tests, we should anyway move into a separate namespace ( `composer-bin` ) to separate the code for acceptance test framework from anything ownCloud core relies on

**Comment by phil-davis** [comment]
ToDo:
- [ ] fix these new tests that seem to use old Guzzle:
```
) DeleteHelperTest::testDeleteHelperWithOlderDavVersion
Error: Class 'GuzzleHttp\Subscriber\Mock' not found

/drone/src/tests/TestHelpers/Unit/DeleteHelperTest.php:41

2) DeleteHelperTest::testDeleteHelperWithNewerDavVersion
Error: Class 'GuzzleHttp\Subscriber\Mock' not found

/drone/src/tests/TestHelpers/Unit/DeleteHelperTest.php:41

3) DeleteHelperTest::testDeleteHelperSendsWithGivenHeaders
Error: Class 'GuzzleHttp\Subscriber\Mock' not found

/drone/src/tests/TestHelpers/Unit/DeleteHelperTest.php:41

4) WebDavHelperTest::testUrlIsSanitizedByMakeDavRequestForNewerDav
Error: Class 'GuzzleHttp\Subscriber\Mock' not found

/drone/src/tests/TestHelpers/Unit/WebDavHelperTest.php:42

5) WebDavHelperTest::testUrlIsSanitizedByMakeDavRequestForOlderDavPath
Error: Class 'GuzzleHttp\Subscriber\Mock' not found
...
```

**Comment by phil-davis** [comment]
Latest hassle: https://github.com/guzzle/guzzle/blob/master/UPGRADING.md e.g. 
`Removed GuzzleHttp\BatchResults`
but acceptance test speedup changes have now used `BatchResults` and sending requests in parallel etc in `UserHelper.php` and `HttpRequestHelper.php`

So to get this Guzzle6 passing again, that has to be sorted out - how to do it when using Guzzle6?

**Comment by DeepDiver1975** [comment]
As discussed in https://github.com/owncloud/core/issues/35777 the master branch will from now on hold the ownCloud 10 codebase. 

This PR targetted ownCloud 11 which is postponed to a far distant future.

Because of that I'm closing this PR and kindly ask you to re-submit this PR in a few days.

Thanks a lot for your patience

**Comment by phil-davis** [comment]
Re-opening so that I can find this some day.

The diffs here are out-of-date due to both the passage of time and the `stable10`->`master` changes. But it will give some clue about what is needed.

**Comment by phil-davis** [comment]
PR #36946 is attempting to update to Guzzle6 for the acceptance test code. That takes the acceptance test changes-in-progress from here and makes them work with current core master.

The changes to actual core backend ccode from here will need to be taken "some day" if we do the "real" core update to Guzzle6.

**Comment by phil-davis** [comment]
PR #36999 implemented Guzzle6 for the acceptance tests.

I will pick out the remaining code changes for the back-end run-time from here and make another PR for that. Using Guzzle6 in core back-end looks like it will be needed for proper PHP 7.4 support.

**Comment by phil-davis** [comment]
See #38706

**Pull request closed without merging**